botorch/acquisition/analytic.py

```diff
@@ -608,7 +607,7 @@ class LogNoisyExpectedImprovement(AnalyticAcquisitionFunction):
 
     def __init__(
         self,
-        model: GPyTorchModel,
+        model: SingleTaskGP,
```

Comment:
Although it's a very ugly hack, I'd suggest leaving this as `SingleTaskGP` and raising an `UnsupportedError` when the model is something other than a `SingleTaskGP`, because this will create a lot of downstream typecheck errors.

Reply:
OK yeah, that's what's currently done in `_get_noiseless_fantasy_model`. And I think that it would be a good idea to make this explicit in the documentation as well. Which is, like you say, hacky since it says that the type is `GPyTorchModel`.

This makes me think of another thing, which is, `SingleTaskGP` can be multiple-outcome, but in `LogNoisyExpectedImprovement` and `NoisyExpectedImprovement` it is assumed to be single-outcome, but there is no explicit check for this.

Reply:
But then, there's the fact that all the other classes in `analytic.py`, and probably most of the BoTorch code, doesn't explicitly check that the models are single-outcome...